force-app/main/default/classes/SOQL.cls

```diff
@@ -16,343 +16,463 @@
+            }
+
+            String searchTerm = '%' + keyword + '%';
+            return [SELECT Id, Title, LeadSource, LastName, Company, Phone, Email
+                    FROM Lead
+                    WHERE (LeadSource = :leadSource
+                    OR Title LIKE :searchTerm)
+                    AND LastName != null
+                    LIMIT 50000];
+        } catch(Exception e) {
+            return new List<Lead>();
+        }
+    }
+
+    /**
+     * Question 11
+     * Returns contact names for an account
+     * @param accountId The Account Id to query contacts for
+     * @return List<String> List of contact last names
+     */
+    public static List<String> getContactNamesByAccount(Id accountId) {
+        List<String> contactNames = new List<String>();
+        for (Contact cont : [SELECT LastName
+                           FROM Contact
+                           WHERE AccountId = :accountId
+                           LIMIT 50000]) {
+            contactNames.add(cont.LastName);
+        }
+        return contactNames;
+    }
+
+    /**
+     * Question 12
+     * Returns total revenue for opportunities in specific stage
+     * @param stage The stage to filter by
+     * @return Decimal The total revenue
+     */
+    public static Decimal getTotalRevenueByStage(String stage) {
+        AggregateResult[] results = [
+            SELECT SUM(Amount) total
+            FROM Opportunity
+            WHERE StageName = :stage
+        ];
+        return (results.isEmpty()) ? 0 : (Decimal)results[0].get('total');
+    }
+
+    /**
+     * Question 13
+     * Returns first 10 accounts ordered by name
+     * @return List<Account> List of accounts
+     */
+    public static List<Account> getFirstTenAccounts() {
+        return [SELECT Id, Name
+                FROM Account
+                ORDER BY Name
+                LIMIT 10];
+    }
+
+    /**
+     * Question 14
+     * Returns high value opportunities
+     * @return List<Opportunity> List of filtered opportunities
+     */
+    public static List<Opportunity> getHighValueOpportunities() {
+        return [SELECT Id, Name, Amount, CloseDate
+                FROM Opportunity
+                WHERE Amount > 100000
+                ORDER BY CloseDate
+                LIMIT 5];
+    }
+
+    /**
+     * Question 15
+     * Returns contacts ordered by birthdate
+     * @return List<Contact> List of filtered contacts
+     */
+    public static List<Contact> getBirthdateContacts() {
+        return [SELECT Id, Name, Birthdate
+                FROM Contact
+                WHERE Birthdate != null
+                ORDER BY Birthdate DESC
+                LIMIT 3];
+    }
+
+    /**
+     * Question 16
+     * Returns cases with offset pagination
+     * @return List<Case> List of filtered cases
+     */
+    public static List<Case> getCasesWithStatus() {
+        return [SELECT Id, Status, CreatedDate
+                FROM Case
+                WHERE Status = 'Open'
```

Comment:
Try to filter by not closed instead of open since there could be many other statuses that are considered open. (ex. New)